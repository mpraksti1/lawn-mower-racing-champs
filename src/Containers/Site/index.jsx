import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Home from '../../Pages/Home';
import Roster from '../../Pages/Roster';
import Player from '../../Pages/Player';
import DefaultPage from '../../Pages/404';
import { logout } from '../../Store/Actions/Auth';
import { createPlayer, deletePlayer, retrievePlayers } from '../../Store/Actions/Player';
import StyledDiv from './styles';

const Site = props => {
  // Redirect away from all non-auth routes if not logged in
  const logoutAndRedirect = path => {
    props.logUserOut();
    props.history.push(path);
  }

  if (!window.localStorage.token) {
    logoutAndRedirect('/auth/register');
  } else if (window.localStorage.token && window.localStorage.expiresIn < Date.now()) {
    logoutAndRedirect('/auth/login');
  }

  const {
    createPlayerFunc,
    retrievePlayersFunc,
    deletePlayerFunc,
    isRequesting,
    players
  } = props;

  return (
    <StyledDiv>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route
          path="/roster"
          render={(props) => (
            <Roster
              {...props}
              players={players}
              retrievePlayers={retrievePlayersFunc}
              deletePlayer={deletePlayerFunc}
            />
          )}
        />
        <Route
          path="/player/new"
          render={(props) => (
            <Player
              {...props}
              createPlayer={createPlayerFunc}
              isRequesting={isRequesting}
            />
          )}
        />
        <Route component={DefaultPage} />
      </Switch>
    </StyledDiv>
  );
};

const mapDispatchToProps = ({
  logUserOut: logout,
  createPlayerFunc: createPlayer,
  deletePlayerFunc: deletePlayer,
  retrievePlayersFunc: retrievePlayers,
})

const mapStateToProps = state => ({
  players: state.players.players,
  isRequesting: state.global.isRequesting,
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Site));
