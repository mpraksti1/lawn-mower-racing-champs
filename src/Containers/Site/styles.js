import styled from 'styled-components';

const StyledDiv = styled.div`
  padding: 0 ${ ({ theme: { baseSizeUnit } }) => `${baseSizeUnit * 2}px ${baseSizeUnit * 2}px`};
`;

StyledDiv.displayName = 'SiteStyled';

export default StyledDiv;