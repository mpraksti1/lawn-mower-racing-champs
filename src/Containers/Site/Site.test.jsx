import React from 'react';
import { shallow } from 'enzyme';
import Site from '.';

describe('Site component', () => {
  it('renders without crashing', () => {
    shallow(<Site />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Site />);

    expect(wrapper).toMatchSnapshot();
  });
});
