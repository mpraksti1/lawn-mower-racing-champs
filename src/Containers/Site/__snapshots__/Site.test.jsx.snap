// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site component matches snapshot 1`] = `<Route />`;
